glean_parser/metrics.py

```diff
@@ -312,7 +312,16 @@ def allowed_extra_keys(self):
     def allowed_extra_keys_with_types(self):
         # Sort keys so that output is deterministic
         return sorted(
-            [(k, v["type"]) for (k, v) in self.extra_keys.items()], key=lambda x: x[0]
+            [(k, v["type"]) for (k, v) in self.extra_keys.items() if "type" in v],
```

Comment:
What happens if `type` is not in `v`? Do we simply skip it or is it accounted for due to `allowed_extra_keys_without_types`? I'm not sure that would ever happen, right?

Reply:
> What happens if type is not in v? 

This function was failing when `type` was not in `v`. I guess it was not called anywhere else, that is why it was not causing trouble before. With the current approach we simply skip it, but jan-erik's comment changes it and is nicer than the current approachm heh so I'll apply it.